Libraries/TranscriptTestRunner/XUnit/XUnitTestRunner.cs

```diff
@@ -36,8 +37,17 @@ public XUnitTestRunner(TestClientBase client, ILogger logger = null)
         /// <returns>A task that represents the work queued to execute.</returns>
         protected override Task AssertActivityAsync(TestScriptItem expectedActivity, Activity actualActivity, CancellationToken cancellationToken = default)
         {
+            var templateRegex = new Regex(@"\{\{[\w\s]*\}\}");
+
             foreach (var assertion in expectedActivity.Assertions)
             {
+                var template = templateRegex.Match(assertion);
+
+                if (template.Success)
+                {
+                    ValidateVariable(template.Value, actualActivity);
+                }
+                
                 var (result, error) = Expression.Parse(assertion).TryEvaluate<bool>(actualActivity);
 
                 Assert.True(result, $"The bot's response was different than expected. The assertion: \"{assertion}\" was evaluated as false.\nActual Activity:\n{JsonConvert.SerializeObject(actualActivity, Formatting.Indented)}");

```

Comment:
We may need to update the branch to the latest in scenariosStaging, this line on that branch shows as follows:

```
                Assert.True(result, $"The bot's response was different than expected. The assertion: \"{assertion}\" was evaluated as false.\nExpected Activity:\n{JsonConvert.SerializeObject(expectedActivity, Formatting.Indented)}\nActual Activity:\n{JsonConvert.SerializeObject(actualActivity, Formatting.Indented)}");

```
Would like to preserve that if possible.

Reply:
Done. Thanks!